### ISSUE TITLE
Not working with Docling - PackageNotFoundError

### ISSUE THREAD

**Issue by drewbroadbent** [Nuitka/Nuitka#3228]
- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.

  > python -m nuitka --version

2.5.4
Commercial: None
Python: 3.12.5 (tags/v3.12.5:ff3bc82, Aug  6 2024, 20:45:27) [MSC v.1940 64 bit (AMD64)]
Flavor: CPython Official
Executable: ~\Environments\NuitkaTest\Scripts\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 11
Version C compiler: ~\AppData\Local\Nuitka\Nuitka\Cache\DOWNLO~1\gcc\x86_64\14.2.0posix-19.1.1-12.0.0-msvcrt-r2\mingw64\bin\gcc.exe (gcc 14.2.0).

- How did you install Nuitka and Python

  Did you use pip, anaconda, deb, rpm, from source, git clone, then install into a virtualenv or

I used pip to install Nuitka + Docling into a virtualenv.

- The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g. `numpy`, you are
  expected to shorten this to the relevant ones.

  > python -m pip list -v

Can all be installed simply by pip install docling + pip install Nuitka

annotated-types           0.7.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
attrs                     24.2.0      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
beautifulsoup4            4.12.3      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
certifi                   2024.8.30   C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
charset-normalizer        3.4.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
click                     8.1.7       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
colorama                  0.4.6       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
deepsearch-glm            0.26.1      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
docling                   2.8.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
docling-core              2.5.1       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
docling-ibm-models        2.0.6       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
docling-parse             2.1.2       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
docutils                  0.21.2      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
easyocr                   1.7.2       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
et_xmlfile                2.0.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
filelock                  3.16.1      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
filetype                  1.2.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
fsspec                    2024.10.0   C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
huggingface-hub           0.26.2      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
idna                      3.10        C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
imageio                   2.36.0      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
Jinja2                    3.1.4       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
jsonlines                 3.1.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
jsonref                   1.1.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
jsonschema                4.23.0      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
jsonschema-specifications 2024.10.1   C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
lazy_loader               0.4         C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
lxml                      5.3.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
markdown-it-py            3.0.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
marko                     2.1.2       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
MarkupSafe                3.0.2       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
mdurl                     0.1.2       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
mpmath                    1.3.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
networkx                  3.4.2       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
ninja                     1.11.1.2    C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
Nuitka                    2.5.4       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
numpy                     1.26.4      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
opencv-python-headless    4.10.0.84   C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
openpyxl                  3.1.5       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
ordered-set               4.1.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
packaging                 24.2        C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
pandas                    2.2.3       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
pillow                    10.4.0      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
pip                       24.2        C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
pyarrow                   16.1.0      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
pyclipper                 1.3.0.post6 C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
pydantic                  2.9.2       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
pydantic_core             2.23.4      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
pydantic-settings         2.6.1       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
Pygments                  2.18.0      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
pypdfium2                 4.30.0      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
python-bidi               0.6.3       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
python-dateutil           2.9.0.post0 C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
python-docx               1.1.2       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
python-dotenv             1.0.1       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
python-pptx               1.0.2       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
pytz                      2024.2      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
pywin32                   307         C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
PyYAML                    6.0.2       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
referencing               0.35.1      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
requests                  2.32.3      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
rich                      13.9.4      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
rpds-py                   0.21.0      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
Rtree                     1.3.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
scikit-image              0.24.0      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
scipy                     1.14.1      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
setuptools                75.6.0      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
shapely                   2.0.6       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
shellingham               1.5.4       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
six                       1.16.0      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
soupsieve                 2.6         C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
sympy                     1.13.1      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
tabulate                  0.9.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
tifffile                  2024.9.20   C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
torch                     2.5.1       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
torchvision               0.20.1      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
tqdm                      4.67.1      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
typer                     0.12.5      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
typing_extensions         4.12.2      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
tzdata                    2024.2      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
urllib3                   2.2.3       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
XlsxWriter                3.2.0       C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip
zstandard                 0.23.0      C:\Users\drewb\Environments\NuitkaTest\Lib\site-packages pip

- Also supply a Short, Self Contained, Correct, Example

from docling.document_converter import DocumentConverter

- Provide in your issue the Nuitka options used

--standalone

- Error:

importlib.metadata.PackageNotFoundError: No package metadata was found for docling-core

**Comment by kayhayen** [comment]
Wow, docling is a dependency monster for sure.

**Comment by kayhayen** [comment]
I added the metadata stuff, then I found that pdfium config only applied to Windows it seems, added a Linux one too, which is where I reproduced the issue, and now rtree seems to miss a DLL, working my way through these, but it seems while some of this is on factory branch right now, it's not going to be complete for 2.5.5 hotfix release.

**Comment by kayhayen** [comment]
The changes are in 2.5.5, however not all docling dependencies are working yet, more work will be needed.

**Comment by drewbroadbent** [comment]
> The changes are in 2.5.5, however not all docling dependencies are working yet, more work will be needed.

Thank you, any idea when this might be fixed (assuming it's still not working in the factory branch)?

**Comment by kayhayen** [comment]
No, I don't give time lines.

**Comment by drewbroadbent** [comment]
Having downloaded and tested the factory build (on windows), I am getting another module not found error: ModuleNotFoundError: No module named 'torch._dynamo.polyfills.builtins'

I am compiling the exact same file as before. Nothing is different, other than build of Nuitka.

**Comment by kayhayen** [comment]
What a masterpiece

```
# See also the TYPE_CHECKING block in torch/_dynamo/polyfills/__init__.py
POLYFILLED_MODULE_NAMES: Tuple[str, ...] = (
    "builtins",
    "functools",
    "itertools",
    "os",
    "sys",
)
POLYFILLED_MODULES: Tuple["ModuleType", ...] = tuple(
    importlib.import_module(f".{submodule}", package=polyfills.__name__)
    for submodule in POLYFILLED_MODULE_NAMES
)
```

**Comment by kayhayen** [comment]
I have solved that and it's also on factory branch and will come to the next hotfix, with that your docling example code works for me now. You never know when it's the last issue, apparently this one was.

**Comment by kayhayen** [comment]
Part of the 2.5.6 hotfix I just made.
